### PR TITLE
minor modifications to primality testing

### PR DESCRIPTION
**Pull request by mynameisvinn**
@mikeivanov thanks for the awesome implementation. ive made some minor changes to is_probably_prime - the most notable is endpoints used for test range, with respect to rabin_miller_witness func.

### CODE REVIEW

File: paillier/primes.py [hunk 1]
```diff
@@ -25,24 +25,41 @@ def rabin_miller_witness(test, possible):
...
-        if possible == i:
-            return True
-        if possible % i == 0:
-            return False
+
+    # linear search through known primes - not necessary but speeds up performance somewhat
+    if possible in SMALLPRIMES:
+        return True
+    
+    # if 'possible' is a multiple of a known prime, then it cant be a prime number
```
Comment (mikeivanov) [inline paillier/primes.py:57]: Typo: "cant"

File: paillier/primes.py [hunk 2]
```diff
@@ -25,24 +25,41 @@ def rabin_miller_witness(test, possible):
...
+    '''Primality testing using Rabin-Miller.
+
+    parameters:
+    -----------
+    @possible: int, representing a random integer to be tested for primality. 
+    @k: int, representing number of tests to conduct. 
+
+    returns:
+    --------
+    @boolean: True if possible is composite, False if it may be prime. (nota bene: this is a probabilistic approach.)
```
Comment (mikeivanov) [inline paillier/primes.py:46]: False if `possible` is definitely composite, True if it may be prime.

File: paillier/primes.py [hunk 3]
```diff
@@ -25,24 +25,41 @@ def rabin_miller_witness(test, possible):
...
-            return True
-        if possible % i == 0:
-            return False
+
+    # linear search through known primes - not necessary but speeds up performance somewhat
+    if possible in SMALLPRIMES:
+        return True
+    
+    # if 'possible' is a multiple of a known prime, then it cant be a prime number
+    if (0 in map(lambda x: possible % x, SMALLPRIMES)):
```
Comment (mikeivanov) [inline paillier/primes.py:58]: if any(possible % x == 0 for x in SMALLPRIMES):
   ...
1) `any` can stop iteration early, so no need to evaluate the whole sequence
2) list comprehensions are easier to read

**Comment by mikeivanov** [comment]
@mynameisvinn -- thanks! Looks good except for a couple of points, please see the comments.

**Comment by estranger** [comment]
i was using and testing this code, then i figured out for key pairs <PublicKey: 44521> <PrivateKey: 44100 423>,<PublicKey: 26569> <PrivateKey: 26244 327>,<PublicKey: 49729> <PrivateKey: 49284 447>,<PublicKey: 18769> <PrivateKey: 18496 275>  etc.  It is not giving same plainttext as encrypted after decryption for few random values, where 1<random value<n. it might be for other key pairs too. is it a bug? or am i doing it in wrong way?
code i was running again and again is 
#!/usr/bin/env python
from paillier.paillier import *
x=3;
i=0;
print"plaint text  = ",x;
priv, pub = generate_keypair(16)
print pub
print priv
for j in range(1,1000):
    cx = encrypt(pub, x);
    z = decrypt(priv, pub, cx)
    if z!=x:
	i=(i+1)
        print" Error ",i

**Pull request closed without merging**